qiskit/utils/run_circuits.py

```diff
@@ -470,18 +476,31 @@ def run_circuits(
     Raises:
         QiskitError: Any error except for JobError raised by Qiskit Terra
     """
+    backend_version = getattr(backend, "version", None)
+    if not isinstance(backend_version, int):
```

Comment:
I remember reading [comments about this line](https://github.com/Qiskit/qiskit-terra/pull/5885/files#r717994650) in #5885 which is also related to all the other comments I have. Should I create an issue about distinguishing these two "version number" so that we will not lose track of what changes to make after the deprecation cycle of `BaseBackend`?

> The cause is in `BaseBackend` `version` is an alias method for `configuration.version` which is an unused field from the iqx api payload format

Reply:
In the meanwhile, is it a good idea to add a comment in this line to briefly explain why we check `backend_version` is not `int` instance?

Reply:
Right, this is needed to still support `BaseBackend` based backends since they have a function named `version`. I'll add a comment about this, but also `BaseBackend` is deprecated and this problem will solve itself in a couple months.

Reply:
Comment added in: https://github.com/Qiskit/qiskit-terra/pull/7563/commits/363e8dd699bd099f54b18201c1217424e5383be8